Modules/Core/Sources/Core/Buffer/Buffer.swift

```diff
@@ -105,6 +108,40 @@ public struct Buffer : RandomAccessCollection {
         }
         
     }
+
+    public func withUnsafeBufferPointer<R>(_ body: (UnsafeBufferPointer<Byte>) throws -> R) rethrows -> R {
+        return try bytes.withUnsafeBufferPointer(body)
+    }
+
+    public mutating func withUnsafeMutableBufferPointer<R>(_ body: (inout UnsafeMutableBufferPointer<Byte>) throws -> R) rethrows -> R {
+        return try bytes.withUnsafeMutableBufferPointer(body)
+    }
+}
+
+extension UnsafeBufferPointer {
+    public init() {
+        self.init(start: nil, count: 0)
+    }
+
+    public init(capacity: Int) {
+        let pointer = UnsafeMutablePointer<Element>.allocate(capacity: capacity)
+        self.init(start: pointer, count: capacity)
+    }
+}
+
+extension UnsafeMutableBufferPointer {
+    public init() {
+        self.init(start: nil, count: 0)
+    }
+
+    public init(capacity: Int) {
```

Comment:
Add a bounds check on `capacity == 0` if it is zero we should first allocate a `UnsafeMutablePointer` with count zero but set it's memory to `nil` this actually ensures that `baseAddress` is not nil but is a valid NULL pointer to send to C API's.